### PR TITLE
Feature: Added Support accepting OTLP via Kafka

### PR DESCRIPTION
**Pull request by metapox**
## Which problem is this PR solving?
- Adds two more recognized encodings, otlp-json and otlp-proto.
- https://github.com/jaegertracing/jaeger/issues/3949

## Short description of the changes
-

### CODE REVIEW

File: cmd/ingester/app/processor/span_processor.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"log"
```
Comment (yurishkuro) [inline cmd/ingester/app/processor/span_processor.go:21]: we don't use log package, only zap

Comment (metapox) [reply cmd/ingester/app/processor/span_processor.go:21]: I see

Tell me about it.
Because span_processor does not hold the logger object of zap,
We need to propagate zap objects from parallel_processor to span_processor.
The interface changes, is this acceptable?

Or should I create a new logger in span_processor as follows?
https://github.com/jaegertracing/jaeger/blob/main/cmd/flags/service.go#L93-L96

Comment (yurishkuro) [reply cmd/ingester/app/processor/span_processor.go:21]: `*zap.Logger` is only created in `main()` and passed through to all components that need it. It's ok to change the component constructor's interface to accept a logger.

**Review by yurishkuro — COMMENTED** [review]
the unit tests here are not simulating a multi-message batch, we need to include that as a test scenario

File: plugin/storage/kafka/writer_test.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ var (
 				RefType: model.ChildOf,
 			},
 		},
-		Flags:     model.Flags(1),
+		Flags:     model.Flags(0),
```
Comment (yurishkuro) [inline plugin/storage/kafka/writer_test.go:50]: not clear why this change is needed

**Comment by yurishkuro** [comment]
closing due to inactivity
